Formatting of the new CW (installation using RAW image) (#3423)
uyuni-project/uyuni-docs: modules/common-workflows/pages/workflow-install-from-raw-image.adoc

[[workflow-install-from-raw-image]]
= Installing {productname} Using RAW Image

{productname} offers various image formats, including a RAW format. 
This guide demonstrates how to set up {productname} Server using the raw image.

== Use Case

RAW images are pre-built, ready-to-use representations of a running operating system. 
Instead of being installed through a traditional installer, they are directly copied to the hard disk of the target host.
RAW image format is flexible and compatible with a wide range of virtualization and deployment tools. 

This guide provides essential information about these pre-configured images.

== Outcome 

Successful completion of this workflow results in successful installation of {productname} Server.

== Step-by-step Instructions

> [WARNING]

Reviewer: Nitpick, I think Warning is too strong here.

```suggestion
[NOTE]
```

====
This workflow is complex and time-consuming.
Make sure that adequate testing is done before deploying the procedure in live environment.
====

.Procedure: Prepare the Prerequisites
[role=procedure]

> . *Download the Image:* Download the architecture-specific [literal]``.raw.xz`` image of SUSE Manager 5.0 Server for your target block size.

Reviewer: ```suggestion
. *Download the Image:* Download the architecture-specific [literal]``.raw.xz`` image of {productname} Server for your target block size.
```

> . *Make Image Accessible:* Copy the image to a web server (preferred), NFS server, or expand the [literal]``.raw`` file to add directly as a disk in a virtual environment (if supported).

Reviewer 2: We should remove the NFS server from this list - as the instructions are more complicated, and copying it to a webserver is not an onerous requirement.  
We should include the 'dd' instructions if we give the option to direct mount the '.raw' disk - as this doc does not yet contain those steps.

Author: @dvosburg - I have removed NFS server from the list according to your instructions, but I need more details about the second part of this comment. What do 'dd' instructions need to look like?

Reviewer 2: If you used NFS as the server type from which to serve the image, use dd to copy it:
For example if the .raw is expanded:
dd if=SUSE-Manager-Server.ppc64le-5.0.0-Raw-ppc-4096-GM.raw of=/dev/sda bs=4M status='progress'
Use dd to copy the compressed raw image to the root disk like this:
xz -d SUSE-Manager-Server.ppc64le-5.0.0-Raw-ppc-4096-GM.raw.xz | dd of=$device bs=4M status='progress'

. *Boot Media:* Provide a bootable ISO ({sles} 15 SPX or {sle-micro} 5.5) to access the "Rescue System." 

>   Add [literal]``netsetup=1`` on the kernel command line for networking, or configure IP via [command]``ifcfg``.

Reviewer: ```suggestion
```

This step is happening during boot, i.e. next step, not during preparation.

> . *Target Disk Setup:* In VMware, create:

Reviewer: This is generic workflow, so no specific hypervisor should be here.

```suggestion
. *Target Disk Setup:* If applicable, in virtual machine hypervisor, create:
```

.. A 40GB root disk
.. Additional storage for persistent SUSE Manager data

											
.Procedure: Installation Steps
[role=procedure]										
. *Prepare Network and Boot Environment*
.. Create a new instance, attach the root and persistent storage, then attach the {sles} or {sle-micro} ISO as a virtual CD.
.. Boot the instance to the ISO, selecting the [literal]``Rescue System``.
.. Add the following on the command line to enable network configuration.
+
----

> netsetup=1

Reviewer: This may work now, but AFAIK SLE16 (and I wonder if SL-Micro 6.1) does not ship linuxrc.

How about using `ip=` from dracut ( https://manpages.opensuse.org/Tumbleweed/dracut/dracut.cmdline.7.en.html#Network )

e.g. `ip=dhcp` for automatic assignment.

Reviewer 2: We are talking about SUSE Manager 5.0 here, and are recommending people boot from SLE15/SLE Micro 5.X  generation rescue media.  The dracut command does not work from the linuxrc offered there.
We are not supporting SL Micro 6.1 nor SLE 16 media for these purposes until a later release of SUSE Manager.  

Quite often customers may not have DHCP available.  By choosing 'netsetup=1' they are prompted to either choose DHCP or configure manual settings.    Alternatively  they can follow the linuxrc 'ifcfg' configuration referenced here:
https://en.opensuse.org/SDB:Linuxrc#Using_ifcfg

Reviewer: Well of course, I was hoping to have something which will work later as well. Still we can recommend SLE15 boot media till 2037, so I guess we are good.

----
+
. *Configure Network and Identify Devices*
.. Complete network setup as prompted and log in as root.
.. Use [command]``fdisk`` to confirm the root device name.
+
----

> fdisk -l

Reviewer: This step can be skipped since we have the same information in the `lsblk` output when we omit device name.

----

> blockdev --getpbsz /dev/sda

Reviewer: Let's use `$device` to be consistent with the curl command later.

```suggestion
blockdev --getpbsz $device
```

----
+
----

> lsblk -o NAME,PHY-SEC /dev/sda

Reviewer: If we skip device name, lsblk will show all and we can choose the right one. This command should be moved up to replace `fdisk -l` command.

```suggestion
lsblk -o NAME,PHY-SEC
```

----
+
. *Transfer the Image*
.. Use [literal]``curl`` to copy the [literal]``.raw.xz`` image onto the root disk:
+
----
curl -Sks $url | xz -d > $device -v
----
+
.. Replace [literal]``$url``  with the image URL and [literal]``$device`` with the root partition device.
. *Finalize Boot and Configuration*
.. Power off the instance, remove the virtual CD from the boot sequence, and start it again to boot from the newly created instance.

> .. Follow on-screen installation directions and, if needed, register with {scc}, adding the SUSE Manager Server Extension.

Reviewer: ```suggestion
.. Follow on-screen installation directions and, if needed, register with {scc}, adding the {productname} Server Extension.
```

. *Install VMware Tools (if applicable)*
.. Install the [literal]``open-vm-tools``
+
----
transactional-update pkg install open-vm-tools
----
+
.. Reboot after installation.
. *Provision Persistent Storage*
+
----
mgr-storage-server $device
----
+

> Ensure [literal]``$device`` points to the fast, persistent storage added in VMware.

Reviewer: ```suggestion
Ensure [literal]``$device`` points to the fast, persistent storage.
```
